### PR TITLE
perf(metrics) Add some metrics to the batching consumer

### PR DESCRIPTION
**Pull request by fpacifici**
We do not have a metric on the number of items in the batch during flush and on the time between two flushes.

### CODE REVIEW

File: arroyo/processing/strategies/batching.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def __init__(
 
         self.__batch: Optional[Batch[TResult]] = None
         self.__closed = False
+        self.__flush_done: Optional[float] = None
```
Comment (lynnagara) [inline arroyo/processing/strategies/batching.py:116]: Should this also be time.time()?

Comment (fpacifici) [reply arroyo/processing/strategies/batching.py:116]: not really. We do not necessarily start submitting messages right after instantiation.

Comment (lynnagara) [reply arroyo/processing/strategies/batching.py:116]: Aren't we just skipping the recording of the first batch then?

For future batches it looks like we are setting this value on the prior call to `strategy.poll()`, which seems to happens pretty much after instantiation, even if no messages were submitted.

Comment (fpacifici) [reply arroyo/processing/strategies/batching.py]: well, this is a good point

**Review by lynnagara — APPROVED** [review]

**Pull request merged**